ui/decredmaterial/editor.go

```diff
@@ -81,88 +88,112 @@ func (t *Theme) Editor(editor *widget.Editor, hint string) Editor {
 			},
 		},
 	}
+	e.TitleLabel.Text = m.Hint
+
+	return e
 }
 
-func (e Editor) Layout(gtx layout.Context) layout.Dimensions {
+func (e *Editor) Layout(gtx layout.Context) layout.Dimensions {
 	e.handleEvents()
 	if e.IsVisible {
 		e.flexWidth = 0.93
 	}
+
 	if e.Editor.Focused() || e.Editor.Len() != 0 {
-		e.TitleLabel.Text = e.Hint
-		e.LineColor = color.RGBA{41, 112, 255, 255}
 		e.Hint = ""
 	}
 
+	if !e.Editor.Focused() {
+		e.Hint = e.StartHint
+	}
+
 	if e.IsRequired && !e.Editor.Focused() && e.Editor.Len() == 0 {
 		e.ErrorLabel.Text = e.requiredErrorText
 	}
 
-	return layout.UniformInset(unit.Dp(2)).Layout(gtx, func(gtx C) D {
-		return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+	if !e.hasCalculated {
```

Comment:
The Layout method seems to do a lot. Add comments to different sections so how it works can be clear.